integrationTests/vm/txsFee/guardAccount_test.go

```diff
@@ -366,28 +458,207 @@ func TestGuardAccount_SendingFundsWhileProtectedAndNotProtected(t *testing.T) {
-	err = transferFundsWithGuardian(testContext, userAddress, big.NewInt(transferValue), receiverAddress, nil)
+	err = transferFundsCoSigned(testContext, userAddress, big.NewInt(transferValue), receiverAddress, nil)
 	require.ErrorIs(t, err, process.ErrTransactionNotExecutable)
 	require.Contains(t, err.Error(), "mismatch between transaction guardian and configured account guardian")
 	require.Equal(t, big.NewInt(transferValue*4), getBalance(testContext, receiverAddress))
 }
+
+// Scenario 1 description:
+// 1.  create & mint 4 addresses: alice, bob, charlie abd delta
+// 2.  alice sets bob as guardian (test if pending)
+// 3.  alice can not set bob as guardian again (test if pending & same activation epoch)
+//   3.1 alice can not set bob as guardian again even if one epoch past
+// 4.  alice activates the guardian (test if active)
+// 5.  alice sets charlie as pending guardian (test if pending & different activation epoch)
+//   5.1. alice wants to set delta as pending guardian (transaction is not executable, will not be included in a miniblock)
+// 6.  alice sets charlie as guardian immediately through a cosigned transaction (test active & pending guardians)
+// 7.  alice immediately sets bob as guardian through a cosigned transaction (test active & pending guardians)
+// 8.  alice adds charlie as a pending guardian (test if pending & different activation epoch)
+//     wait until charlie becomes active, no more pending guardians
+// 9.  alice adds bob as a pending guardian and calls set charlie immediately cosigned and should remove the pending guardian
+// 10. alice un-guards the account immediately by using a cosigned transaction
+// 11. alice guards the account immediately by calling the GuardAccount function
+// 13. alice sends a guarded transaction, while account is guarded -> should work
+// 14. alice un-guards the accounts immediately using a cosigned transaction and then sends a guarded transaction -> should error
+//   14.1 alice sends unguarded transaction -> should work
+func TestGuardAccount_Scenario1(t *testing.T) {
+	testContext := prepareTestContextForFreezeAccounts(t)
+	defer testContext.Close()
+
+	uuid := []byte("uuid")
+	// transferValue := int64(2000000)
+	initialMint := big.NewInt(1000000000000000000)
+
+	alice := []byte("alice-12345678901234567890123456")
+	bob := []byte("bob-1234567890123456789012345678")
+	charlie := []byte("charlie-123456789012345678901234")
+	delta := []byte("delta-12345678901234567890123456")
+
+	// step 1 -  mint addresses
+	allAddresses := [][]byte{alice, bob, charlie}
+	for _, address := range allAddresses {
+		mintAddress(t, testContext, address, initialMint)
+	}
+	for _, address := range allAddresses {
+		testNoGuardianIsSet(t, testContext, address)
+	}
+	currentEpoch := uint32(0)
+
+	// step 2 - alice sets bob as guardian
+	step2Epoch := currentEpoch
+	returnCode, err := setGuardian(testContext, alice, bob, uuid)
+	assert.Nil(t, err)
```

Comment:
maybe you can use `require` instead of `assert`  in order to stop the test execution when something will go wrong